### PR TITLE
Write basic parser for comment references

### PR DESCRIPTION
**Pull request by jcollins-g**
This is intended to replace the string splits/joins and regular expressions scattered throughout markdown_processor.dart in the new lookup system.

### CODE REVIEW

File: lib/src/comment_references/parser.dart [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+      } else {
+        return;
+      }
+    }
+    return;
+  }
+
+  /// Returns true if we started with startChar and ended with endChar
+  /// with a matching number of braces.
+  /// Returns _index to start if returning false.
```
Comment (srawlins) [inline lib/src/comment_references/parser.dart:224]: "startIndex" would be more meaningful than "start" here.

**Review by srawlins — APPROVED** [review]
Excellent! Impl and tests lgtm

File: lib/src/comment_references/parser.dart (hunk 1 above)
Comment (jcollins-g) [reply lib/src/comment_references/parser.dart:224]: rephrased

File: lib/src/comment_references/parser.dart [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+      if (_whitespace.contains(_thisChar)) {
+        _index++;
+      } else {
+        return;
+      }
+    }
+    return;
+  }
+
+  /// Returns true if we started with startChar and ended with endChar
```
[the remark this answers is not in the document]
Comment (jcollins-g) [reply lib/src/comment_references/parser.dart:222]: done

File: lib/src/comment_references/parser.dart [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+    // <prefix>?
+    var prefixResult = _parsePrefix();
+    if (prefixResult.type == _PrefixResultType.endOfFile) {
+      return [];
+    }
+    if (prefixResult.type == _PrefixResultType.parsedConstructorHint) {
+      children.add(prefixResult.node);
+    }
+    // junk / missing we can skip.
```
[the remark this answers is not in the document]
Comment (jcollins-g) [reply lib/src/comment_references/parser.dart:46]: done

**Pull request merged**